public/electron.js

```diff
@@ -35,8 +37,73 @@ function createWindow() {
     mainWindow.show();
   });
   ipcHandler(mainWindow);
+
+  mainWindow.on("minimize", (e) => {
+    e.preventDefault();
+    mainWindow.hide();
+    tray.setContextMenu(trayMenuWithShow);
+  });
+
+  mainWindow.on("close", (e) => {
+    if (!app.isQuiting) {
+      e.preventDefault();
+      mainWindow.hide();
+      tray.setContextMenu(trayMenuWithShow);
```

Comment:
A minor nit: could you please move this logic into a smaller function that could be called here and for `minimize` as well? Would make the code DRYer.